site/js/ledAPI.js

```diff
@@ -0,0 +1,256 @@
+                nameMaxLength));
+            document.getElementById(deviceID).value = input.slice(0,
+                nameMaxLength);
+
+        });
+
+    }
+
+    function updateOnButton(device) {
+        console.debug("updateOnButton: state received: ", device[1][
+            'onState'
+        ]);
+        switch (device[1]['onState']) {
+            case "true":
+                $("#onSwitch").prop("checked", true);
+                break;
+            case "false":
+                $("#onSwitch").prop("checked", false);
+                break;
+        }
+    }
+
+
+    function updateBrightSlider(device) {
+        var brightVal = device[1]['brightness']
+        $('#brightness-value').html(brightVal);
+        $('#slider-brightness').val(brightVal);
+    }
+
+    function updateDeviceList(newDevices) {
+        var remove = [];
+        var add = [];
+
+        newDevices.sort();
+
+        /* Identify old devices no longer online and remove */
+        for (var i in devices) {
+            if (newDevices.indexOf(devices[i]) === -1) remove.push(devices[
+                i]);
+        }
+        remove.forEach(removeDevice);
+
+        /* Hide parameter UI and current device label if not online (e.g., no heartbeat) */
+        if (newDevices.includes(("hb_" + deviceID)) == false) {
+            $("#currentDeviceLabel").hide();
+            $("#parameter-UI").toggleClass('look-disabled', true);
+            console.debug("hiding currentDeviceLabel");
+        } else {
+            console.debug(
+                "getting/showing currentDeviceLabel and parameters");
+
+            /* if current device is in heartbeat list, get device info */
+            $apiUtils.getData(apiUrl, deviceID, updateMainContent,
+                disableMainContent);
+
+            $("#currentDeviceLabel").show();
+            $("#parameter-UI").toggleClass('look-disabled', false);
+        }
+
+        /* Identify new devices received in order to add */
+        for (var i in newDevices) {
+            if (devices.indexOf(newDevices[i]) === -1) add.push(newDevices[
+                i]);
+        }
+        add.forEach(insertDevice);
+
+        /* update device list */
+        devices = newDevices;
+
+        $('#main-content').toggleClass('look-disabled', false);
+    }
+
+    async function insertDevice(device, index) {
+        device = device.slice(3, device.length)
+        console.debug("device to insert: ", device);
+
+        /* Try to retrieve device name from server. 
+         * If unable to, log error and set device name to shortened deviceID */
+        try {
+            console.debug("getting device name to insert");
+            var deviceName = await $apiUtils.getParam(apiUrl, device,
+                "name");
+        } catch (error) {
+            console.error("Error caught: ", error);
+            var deviceName = device.slice(7, 15); // if device doesn't have a name already, create shortened name from ID
+        }
```

Comment:
This gets a bit confusing since `var deviceName` is defined twice in the same function.  I suspect some lint tools will call this out.

Also: this is one of the places where `let` might actually be cleaner.

Since `let` is kinda like `var`, but "block scoped" rather than "function scoped" it forces you to pull the variable detection into the the outer block anyway.

```suggestion
       let deviceName;
       try {
            console.debug("getting device name to insert");
            deviceName = await $apiUtils.getParam(apiUrl, device,
                "name");
        } catch (error) {
            console.error("Error caught: ", error);
            deviceName = device.slice(7, 15); // if device doesn't have a name already, create shortened name from ID
        }
```

Reply:
good suggestion. thanks :)